### PR TITLE
ddl: fix `DROP [GLOBAL] TEMPORARY TABLE IF EXISTS` returns error when table not exist

### LINKED ISSUE #27276: `Drop if exists` did not work well for temporary table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
drop global temporary table if exists test.noexist;
```

```
drop temporary table if exists test.noexist;
```

```
create table test.t1 (a int);
drop temporary table if exists test.t1;
```

### 2. What did you expect to see? (Required)

```
mysql> drop global temporary table if exists test.noexist;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```

```
mysql> drop temporary table if exists test.noexist;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```

```
mysql> drop temporary table if exists test.t1;
Query OK, 0 rows affected, 1 warning (0.00 sec)
```

### 3. What did you see instead (Required)

```
mysql> drop global temporary table if exists test.noexist;
ERROR 1051 (42S02): Unknown table 'test.noexist'
```

```
mysql> drop temporary table if exists test.noexist;
ERROR 1051 (42S02): Unknown table 'test.noexist'
```

```
mysql> drop temporary table if exists test.t1;
ERROR 1051 (42S02): Unknown table 'test.t1'
```

### 4. What is your TiDB version? (Required)

master

### PR DESCRIPTION
**Pull request by lcwangchao**
… table not exist

### What problem does this PR solve?

Issue Number: close #27276 

Problem Summary:

`DROP [GLOBAL] TEMPORARY TABLE IF EXISTS` still produce error when table not exist.

### What is changed and how it works?

Remove logic to handle table not exists in preprocessor.

### Check List 

Tests 

- [x] Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

File: ddl/db_integration_test.go [hunk 1]
```diff
@@ -3162,6 +3162,163 @@ func (s *testIntegrationSuite3) TestDropTemporaryTable(c *C) {
...
+	c.Assert(infoschema.ErrTableDropExists.Equal(err), IsTrue)
+
+	// testing for drop table if exists which is not local temporary
+	tk.MustExec("drop temporary table if exists xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.xxx'"))
+	tk.MustExec("drop temporary table if exists ltemp1, xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.xxx'"))
+	tk.MustExec("drop temporary table if exists tb1, xxx")
+	tk.MustQuery("show warnings").Check(testkit.Rows("Note 1051 Unknown table 'test.tb1,test.xxx'"))
+	tk.MustExec("drop temporary table if exists temp1, xxx")
```
Comment (djshow832) [inline ddl/db_integration_test.go:3288]: What if `drop temporary table if exists temp1`? We don't allow it.

Comment (lcwangchao) [reply ddl/db_integration_test.go:3288]: I'll add these case later

Comment (lcwangchao) [reply ddl/db_integration_test.go:3290]: Done

**Review by djshow832 — APPROVED** [review]

File: ddl/db_integration_test.go [hunk 2]
```diff
@@ -3162,6 +3162,165 @@ func (s *testIntegrationSuite3) TestDropTemporaryTable(c *C) {
...
+	tk.MustExec(clearSQL)
+	defer tk.MustExec(clearSQL)
+	// two normal table test.tb, test.tb2, a temporary table with name test.tb2
+	tk.MustExec("create table tb(id int)")
+	tk.MustExec("create table tb2(id int)")
+	tk.MustExec("insert into tb2 values(1)")
+	tk.MustExec("create temporary table tb2(id int)")
+	// two global temporary table test.temp, test.temp1
+	tk.MustExec("create global temporary table temp(id int) on commit delete rows")
+	tk.MustExec("create global temporary table temp1(id int) on commit delete rows")
```
Comment (tiancaiamao) [inline ddl/db_integration_test.go:3255]: From the test function name `TestDropLocalTemporaryTableWithDropTemporaryStmt`  this is a bit confusing

File: ddl/db_integration_test.go [hunk 3]
```diff
@@ -3162,6 +3162,165 @@ func (s *testIntegrationSuite3) TestDropTemporaryTable(c *C) {
...
+	defer tk.MustExec(clearSQL)
+	// two normal table test.tb, test.tb2, a temporary table with name test.tb2
+	tk.MustExec("create table tb(id int)")
+	tk.MustExec("create table tb2(id int)")
+	// two global temporary table test.temp, test.temp1
+	tk.MustExec("create global temporary table temp(id int) on commit delete rows")
+	tk.MustExec("create global temporary table temp1(id int) on commit delete rows")
+	// two local temporary table test.ltemp1, test.ltemp2
+	tk.MustExec("create temporary table ltemp1(id int)")
+	tk.MustExec("create temporary table ltemp2(id int)")
```
Comment (tiancaiamao) [inline ddl/db_integration_test.go:3181]: In function `TestDropGlobalTemporaryTableWithDropGlobalStmt`, local temporary table is ... covered.

File: planner/core/preprocess_test.go [hunk 4]
```diff
@@ -376,47 +376,5 @@ func (s *testValidatorSuite) TestDropGlobalTempTable(c *C) {
 	s.runSQL(c, "drop global temporary table temp, temp1", false, nil)
 	s.runSQL(c, "drop global temporary table temp, tb", false, core.ErrDropTableOnTemporaryTable)
 	s.runSQL(c, "drop global temporary table temp, ltemp1", false, core.ErrDropTableOnTemporaryTable)
-	s.runSQL(c, "drop temporary table xxxx, ltemp1", false, infoschema.ErrTableDropExists)
 	s.runSQL(c, "drop global temporary table test2.temp2, temp1", false, nil)
 }
-
-func (s *testValidatorSuite) TestDropLocalTempTable(c *C) {
```
Comment (tiancaiamao) [inline planner/core/preprocess_test.go:383]: Well, the old code style is actually more reader friendly...

**Review by tiancaiamao — APPROVED** [review]

File: planner/core/preprocess_test.go (hunk 4 above)
Comment (tiancaiamao) [reply planner/core/preprocess_test.go:383]: Never mind ... just the coding style and inaccurate comment.
You are doing well enough, good job! @lcwangchao

File: ddl/db_integration_test.go (hunk 2 above)
Comment (lcwangchao) [reply ddl/db_integration_test.go:3255]: This is copied, so..

File: ddl/db_integration_test.go (hunk 3 above)
Comment (lcwangchao) [reply ddl/db_integration_test.go:3181]: This is useful to test dropping a local temporary table with statement `drop global temporary table...`

File: ddl/db_integration_test.go [hunk 5]
```diff
@@ -3162,6 +3162,165 @@ func (s *testIntegrationSuite3) TestDropTemporaryTable(c *C) {
...
 
+func (s *testIntegrationSuite3) TestDropGlobalTemporaryTableWithDropGlobalStmt(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec("set tidb_enable_noop_functions=true")
+	tk.MustExec("set tidb_enable_global_temporary_table=true")
+	clearSQL := "drop table if exists tb, tb2, temp, temp1, ltemp1, ltemp2"
+	tk.MustExec(clearSQL)
+	defer tk.MustExec(clearSQL)
+	// two normal table test.tb, test.tb2, a temporary table with name test.tb2
```
[the remark this answers is not in the document]
Comment (lcwangchao) [reply ddl/db_integration_test.go]: It is copied from another case, so...

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**